code/game/objects/effects/spawners/lootdrop.dm

```diff
@@ -240,18 +254,52 @@
 
 /obj/effect/spawner/lootdrop/maintenance
 	name = "maintenance loot spawner"
+	desc = "Come on Lady Luck, spawn me a pair of sunglasses."
+	spawn_on_init = FALSE
 	// see code/_globalvars/lists/maintenance_loot.dm for loot table
 
+/obj/effect/spawner/lootdrop/maintenance/examine(mob/user)
+	. = ..()
+	. += span_info("This spawner has an effective loot count of [get_effective_lootcount()].")
+
 /obj/effect/spawner/lootdrop/maintenance/Initialize(mapload)
+	. = ..()
+	// There is a single callback in SSmapping to spawn all delayed maintenance loot
+	// so we don't just make one callback per loot spawner
+	GLOB.maintenance_loot_spawners += src
 	loot = GLOB.maintenance_loot
 
+	// Late loaded templates like shuttles can have maintenance loot
+	if(SSticker.current_state >= GAME_STATE_SETTING_UP)
+		spawn_loot()
+		hide()
+
+/obj/effect/spawner/lootdrop/maintenance/Destroy()
+	GLOB.maintenance_loot_spawners -= src
+	. = ..()
+
+/obj/effect/spawner/lootdrop/maintenance/proc/hide()
+	invisibility = INVISIBILITY_OBSERVER
+	alpha = 100
+
+/obj/effect/spawner/lootdrop/maintenance/proc/get_effective_lootcount()
+	var/effective_lootcount = lootcount
+
 	if(HAS_TRAIT(SSstation, STATION_TRAIT_FILLED_MAINT))
-		lootcount = FLOOR(lootcount * 1.5, 1)
+		effective_lootcount = FLOOR(lootcount * 1.5, 1)
 
 	else if(HAS_TRAIT(SSstation, STATION_TRAIT_EMPTY_MAINT))
-		lootcount = FLOOR(lootcount * 0.5, 1)
+		effective_lootcount = FLOOR(lootcount * 0.5, 1)
 
-	. = ..()
+	return effective_lootcount
+
+/obj/effect/spawner/lootdrop/maintenance/spawn_loot()
+	. = ..(get_effective_lootcount())
```

Comment:
```suggestion
/obj/effect/spawner/lootdrop/maintenance/spawn_loot(lootcount_override)
	if(!isnull(lootcount_override))
		lootcount_override = get_effective_lootcount()
	return ..()
```
This makes it more clear what is happening (the proc args get passed down the line) and allows admins to use the verb with a custom argument.

Reply:
But this also triggers the closet taking-contents on lines 299-302, which needs to happen after calling parent.

Can I keep the `. = ..(lootcount_override)`

Reply:
Oh, I missed that part. Let me suggestion edited.